generator/src/lib.rs

```diff
@@ -964,7 +963,20 @@ fn generate_function_pointers<'a>(
             params
         })
         .collect();
-    let signature_params_ref = &signature_params;
+    assert_eq!(pfn_names.len(), pfn_signature_params.len());
+
+    let pfn_return_types: Vec<_> = commands_pfn
+        .iter()
+        .map(|cmd| {
+            if cmd.return_type.is_void() {
+                quote!()
+            } else {
+                let ret_ty_tokens = cmd.return_type.type_tokens(true);
+                quote!(-> #ret_ty_tokens)
+            }
+        })
+        .collect();
```

Comment:
I really hate having to duplicate this (just like the rest of the code here). Perhaps we should pre-generate a struct with all bits, and a boolean whether it needs a typedef created for it.

Reply:
Had a go at this in 0d0d221, let me know what you think! That refactor prevents this class of array-content-mismatches in generation entirely.

Reply:
Applied and pushed the refactor to this PR where it belongs - can also do it as a separate PR if you prefer (before or after #438).

Any reason why we were manually generating a `Clone` implementation instead of deriving it? The latter is easier and saves some code, and does not appear to be slower within margin of error:

```
   Completed ash v0.32.0 in 5.9s
```
Versus manual `impl Clone` for every pointer-struct:
```
   Completed ash v0.32.0 in 5.8s
```

Reply:
> Had a go at this in 0d0d221, let me know what you think! That refactor prevents this class of array-content-mismatches in generation entirely.

Looks good

> Any reason why we were manually generating a Clone implementation instead of deriving it? The latter is easier and saves some code, and does not appear to be slower within margin of error:

I am not sure actually, lets just autoderive it.